hudi-spark/src/main/scala/org/apache/hudi/AvroConversionHelper.scala

```diff
@@ -324,7 +323,7 @@ object AvroConversionHelper {
           avroSchema,
           valueType,
           structName,
-          getNewRecordNamespace(valueType, recordNamespace, structName))
+          recordNamespace)
```

Comment:
Might be good to add a test for Map types as well. We can be sure that we have fixed these issues once and for all, for all the major types we support. Also in future if spark-avro changes something we can immediately fix these issues.

Reply:
Good idea. A test case for Map is added.